src/scvi128.js

```diff
@@ -0,0 +1,100 @@
+import xdr from "./xdr";
+
+/**
+ * @param {bigint} value
+ * @returns {xdr.ScVal}
+ */
+export function bigintToScvI128(value) {
+  const hex = value.toString(16).replace(/^-/, "");
+  if (hex.length > 32) {
+    throw new Error("value overflow i128");
+  }
+
+  const buf = Buffer.alloc(16);
+  buf.write(hex, 16 - Math.ceil(hex.length / 2), "hex"); // BE
+
+  // perform two's compliment if negative
+  if (value < 0) {
+    // throw if MSB bit is 1 and is not i128:MIN
+    if ((buf[0] & 0x80) != 0 && hex != "80000000000000000000000000000000") {
```

Comment:
there must be a better way than this string comparison :sweat_smile: 

what about some equivalent of `value != (1 << 64)`?

Reply:
There likely is, yes! (and probably many more improvements that can be made)

The main goal here was to find when the input value was equivalent to `i128::MIN`, cause any other number with the 128th bit as 1 would be out of range. If `number` can support this with accuracy, I'd imagine `value  != BigInt(1 << 127)` would work.